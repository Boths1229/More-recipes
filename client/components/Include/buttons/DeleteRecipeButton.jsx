import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description functional component to render delete button for recipe
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */
const DeleteRecipeButton = (props) => {
  const { recipeId } = props;
  return (
    <div>
      <button
        type="button"
        className="btn btn-danger ml-3"
        data-toggle="modal"
        data-target={`#deleteModal-${recipeId}`}
        data-whatever="@getbootstrap"
      >
        <i className="fa fa-times" aria-hidden="true" />
      </button>
    </div>
  );
};

DeleteRecipeButton.propTypes = {
  recipeId: PropTypes.number.isRequired
};

export default DeleteRecipeButton;
