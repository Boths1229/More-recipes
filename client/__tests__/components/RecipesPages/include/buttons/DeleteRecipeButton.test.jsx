import React from 'react';
import { shallow } from 'enzyme';
import DeleteRecipeButton from '../../../../../components/Include/buttons/DeleteRecipeButton';

describe('Delete recipe button component', () => {
  it('matches snapshot', () => {
    const output = shallow(<DeleteRecipeButton />);
    expect(output).toMatchSnapshot();
  });
});
