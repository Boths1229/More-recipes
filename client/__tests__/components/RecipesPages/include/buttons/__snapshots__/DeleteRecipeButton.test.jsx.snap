// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete recipe button component matches snapshot 1`] = `
<div>
  <button
    className="btn btn-danger ml-3"
    data-target="#deleteModal-undefined"
    data-toggle="modal"
    data-whatever="@getbootstrap"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-times"
    />
  </button>
</div>
`;
